internal/exec/util/unit.go

```diff
@@ -151,6 +153,22 @@ func (ut Util) EnableUnit(enabledUnit string) error {
 }
 
 func (ut Util) DisableUnit(disabledUnit string) error {
+	// We need to delete any enablement symlinks for a unit before sending it to a
+	// preset directive. This will help to disable that unit completely.
+	// For more information: https://github.com/coreos/fedora-coreos-tracker/issues/392
+	// This is a short-term solution until the upstream systemd PR
+	// (https://github.com/systemd/systemd/pull/15205) gets accepted.
+	if err := ut.Logger.LogOp(
+		func() error {
+			if _, err := exec.Command(distro.SystemCtl(), "--root=/sysroot", "disable", disabledUnit).CombinedOutput(); err != nil {
```

Comment:
Does this correctly handle units which do not exist? (I.e. we shouldn't be erroring out in that case.)

Reply:
Yeah, the code doesn't error out though. I didn't find any warning(s)/errors in journal entries.

Reply:
> Would be good to add a test verifying this too (i.e. that we don't error out if trying to disable a non-existent unit).

Does this come under a negative test?

Reply:
I think putting it in the same positive test you're creating should be fine.